core/solver/gmres.cpp

```diff
@@ -414,6 +427,9 @@ void Gmres<ValueType>::apply_dense_impl(const VectorType* dense_b,
     auto krylov_bases_small = create_submatrix_helper(
         krylov_bases, dim<2>{num_rows, num_rhs},
         span{0, local_num_rows * (restart_iter + 1)}, span{0, num_rhs});
+    auto preconditioned_krylov_bases_small = create_submatrix_helper(
```

Comment:
same here

Reply:
I move the corresponding into the scope because they are only needed once and read-only